ntoskrnl/kd/kdio.c

```diff
@@ -256,6 +258,8 @@ KdpDebugLogInit(PKD_DISPATCH_TABLE DispatchTable,
     KPRIORITY Priority;
     SIZE_T MemSizeMBs;
 
+    ASSERT(ExGetPreviousMode() == KernelMode);
```

Comment:
Useless since anyone understanding the code would know that `KdpDebugLogInit` is called by `KdInitSystem`, which is only called from the kernel and by the kernel, so it's definitively running kernel mode context only.

Reply:
> it's definitively running kernel mode context only.

Obviously. This is not about how current code behaves.
Your point of view is clear. If Thomas doesn't want the asserts either, I'll simply remove them.

> anyone understanding the code

Hum... Could you tell me why, for example,
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=KdbpCliInit
uses `Zw*()`?

Reply:
Eventually, 7caf9e9 does what I initially submitted...